doc/user/content/ingest-data/_index.md

```diff
@@ -0,0 +1,244 @@
+operation is complete, you can downsize your source cluster to align with the
+volume of changes being replicated from your upstream in steady-state.
+{{% /tip %}}
+
+### Queries during snapshotting
+
+While a source is snapshotting, the source (and the associated subsources)
+cannot serve queries. That is, queries issued to the snapshotting source (and
+its subsources) will return after the snapshotting completes (unless the user
+breaks out of the query). If the user does not break out of the query, the
+returned query results will reflect the data from the snapshot.
+
+## Running/steady-state
+
+Once snapshotting completes, Materialize transitions to Running state. During
+this state, Materialize continually ingests changes from the upstream system.
+
+### Queries during steady-state
+
+Although Materialize is continually ingesting changes from the upstream system,
+depending on the volume of the upstream changes, Materialize may lag behind the
+upstream system. If the lag is significant, queries may block until Materialize
+has caught up sufficiently with the upstream system when using the default
+[isolation level](/get-started/isolation-level/) of [strict
+serializability](/get-started/isolation-level/#strict-serializable).
+
+In the Materialize Console, you can see a source's data freshness from the
+**Data Explorer** screen. Alternatively, you can run a query to monitor the lag.
+See [Monitoring the snapshotting progress](/ingest-data/monitoring-data-ingestion).
+
+## Rehydration
+
+When a cluster is restarted (such as after resizing), sources undergo
+rehydration.[^1] Rehydration refers to the reconstruction of in-memory state by
+reading data from the storage layer; rehydration does not require reading data
+from the upstream system.
+
+{{% tip %}}
+
+If possible, use a dedicated cluster just for sources. That is, avoid
+using the same cluster for sources and other objects, such as sinks, etc. See [Best practices](#best-practices) for more details.
+
+{{% /tip %}}
+
+### Process
+
+During rehydration, data from the storage layer is read to reconstruct the
+in-memory state of the object. As part of the rehydration process:
+
+- Internal data structures are re-created.
+
+- Various processes are re-initiated. These processes may also require
+  re-reading of their in-memory state.
+
+### Duration
+
+For a source, the duration of its rehydration depends on the type and the size
+of the source; e.g., large `UPSERT` sources can take hours to complete.
+
+### Queries during rehydration
+
+During rehydration, queries usually block until the process has been completed.
+
+## Best practices
+
+#### Scheduling
+
+If possible, schedule creating new sources during off-peak hours to mitigate the
+impact of snapshotting on both the upstream system and the Materialize cluster.
+
+#### Dedicate a cluster for the sources
+
+If possible, dedicate a cluster just for sources. That is, avoid using the same
```

Comment:
for future: even within the umbrella of dedicated source clusters, further division with dedicated upsert source clusters.

Reply:
> So, you have a source cluster and it's sized for current steady state. What is the happy route for adding a new source to that cluster.

This is why I think we should think about special casing upsert sources. We don't strictly need to scale up clusters with OLTP sources to add a new subsource. And if you do want to scale up it's very fast. 

Though, once 0dt source rescaling lands some of this problem may just go away.